src/overlay/TxDemandsManager.cpp

```diff
@@ -281,7 +307,38 @@ TxDemandsManager::recvTxDemand(FloodDemand const& dmd, Peer::pointer peer)
                        KeyUtils::toShortString(peer->getPeerID()));
             peer->getPeerMetrics().mMessagesFulfilled++;
             om.mMessagesFulfilledMeter.Mark();
-            peer->sendMessage(tx->toStellarMessage());
+
+#ifdef BUILD_TESTS
+            if (mApp.getConfig().EXPERIMENTAL_TX_BATCH_MAX_SIZE > 0)
+            {
+                // Current batch size
+                auto currSize = FlowControlCapacity::msgBodySize(*msg);
+                // New tx size to append
+                auto newTxSize =
+                    FlowControlCapacity::msgBodySize(*tx->toStellarMessage());
+
+                // Send existing batch if it exceeds the max size, create a new
+                // message
+                if ((currSize + newTxSize) <= mApp.getHerder().getMaxTxSize())
+                {
+                    sendAndReset(msg);
+                }
```

Comment:
It looks like this sends the message if it's less than the max size, not greater than as the comment suggests. Unless I'm misreading the comment (or there's supposed to be a period between "batch" and "if").

Reply:
Good catch, I added this last-minute check, and looks like I accidentally pushed a broken test that doesn't do anything. Will fix the condition and the accompanying test.